app/components/Contacts/Network/BrowseChannelButton.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import ExternalLink from 'components/Icon/ExternalLink'
+import { Button } from 'components/UI'
+
+const BrowseChannelButton = ({ isVisible, onClick }) =>
+  isVisible && (
```

Comment:
Why add the `isVisible` prop to each component? It seems a non-reacty thing to do. Shouldn't it be the parent sets wether the prop renders of not in its jsx markup?

```jsx
{isVisible && <BrowseChannelButton />}
```

The `isVisible` seems redundant

Reply:
Hm, maybe you are right, but I find
`{isVisible && <BrowseChannelButton />}`
Not so readable as 
`<BrowseChannelButton isVisible={isVisible}/>`
because of those curly braces
I understand your concern that it's like a black box and you don't know how this flag is handled by each component. Is this what concerns you?

Reply:
Yes, pretty much. It requires the user of the component to know about the prop, what it is and what it does. It means that readers of the parent component don't inherently know that one of it's children will not be rendered because all you can see is that it is included in the markup and has a bunch of props - but what do those props do?

With jsx it is immediately obvious to the reader what will render and what will not and there is no guesswork.

It's also a bit inconsistent for us to handle visibility in this way for these components when we don't do that anywhere else in the codebase.

Reply:
Ok, I understand your point. I will extract those props back